### PR TITLE
docs: Update HA Terraform reference and add starter cluster reference

### PR DESCRIPTION
**Pull request by webvictim**
- Updates docs for HA AWS deployments (which accompany the `ha-autoscale-cluster` Terraform)
  - Updates its README and outputs to make management easier
  - Adds a bash script for quick connection to instances behind the bastion
  - Renames from `aws-terraform` to `aws-ha-autoscale-cluster-terraform` to disambiguate

- Adds docs for an AWS single instance deployment (which accompany the `starter-cluster` Terraform)
  - Updates its README and outputs

- Updates old links in the test plan and issue templates to HA deployment docs

All other changes are my editor trimming whitespace.

### CODE REVIEW

**Comment by webvictim** [comment]
Reminder to self: squash this before merge.

File: docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx [hunk 1]
```diff
@@ -0,0 +1,714 @@
+---
+title: Teleport Single Instance deployment on AWS
+description: How to quickly configure Teleport on a single instance for testing in AWS.
+tocDepth: 3
+---
+
+This guide is designed to accompany our [reference starter-cluster Terraform code](https://github.com/gravitational/teleport/tree/branch/v(=teleport.major_version=)/examples/aws/terraform/starter-cluster#teleport-terraform-aws-ami-simple-example)
```
Comment (ptgott) [inline docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx:7]: I would briefly describe the infrastructure components that this Terraform module deploys so a user can determine whether the module and this guide are appropriate for their use case.

File: docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx [hunk 2]
```diff
@@ -0,0 +1,714 @@
...
+Fedora/CentOS: `yum -y install awscli`
+
+Ubuntu/Debian: `apt-get -y install awscli`
+
+MacOS (with [Homebrew](https://brew.sh/)): `brew install awscli`
+
+When possible, installing via a package is always preferable. If you can't find a package available for
+your distribution, you can also download the tool from [https://aws.amazon.com/cli/](https://aws.amazon.com/cli/)
+
+We will assume that you have configured your AWS cli access with credentials available at `~/.aws/credentials`:
```
Comment (ptgott) [inline docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx:36]:
```suggestion
We will assume that you have configured your AWS CLI access with credentials available at `~/.aws/credentials`:
```

File: docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx [hunk 3]
```diff
@@ -0,0 +1,714 @@
...
+We maintain an up-to-date list of the variables and what they do in the README.md file under [the
+`examples/aws/terraform/ha-autoscale-cluster` section of the Teleport repo](https://github.com/gravitational/teleport/blob/branch/v(=teleport.major_version=)/examples/aws/terraform/starter-cluster/README.md)
+but we'll run through an example list here.
+
+Things you will need to decide on:
+
+### region
+
+```code
+export TF_VAR_region="<Var name="us-west-2" />"
```
Comment (ptgott) [inline docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx:139]: All of the `code` snippets on this page with example commands should start with `$` so the docs engine renders them properly

File: docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx [hunk 4]
```diff
@@ -0,0 +1,714 @@
...
+
+### license_path
+
+(!docs/pages/includes//enterprise/obtainlicense.mdx!)
+
+```code
+export TF_VAR_license_path="/home/user/license.pem"
+```
+
+This license will be uploaded to AWS SSM and automatically downloaded to Teleport auth nodes in order to enable
```
Comment (ptgott) [inline docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx:209]:
```suggestion
This license will be uploaded to AWS SSM and automatically downloaded to Teleport Auth Service nodes in order to enable
```

File: docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx [hunk 5]
```diff
@@ -0,0 +1,714 @@
...
+Teleport Enterprise/Pro functionality.
```
Comment (ptgott) [inline docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx:210]:
```suggestion
Teleport Enterprise functionality.
```

File: docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx [hunk 6]
```diff
@@ -0,0 +1,714 @@
...
+
+(OSS users can provide any valid local file path here - it isn't used by the auth server in a Teleport OSS install)
```
Comment (ptgott) [inline docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx:212]: I would leave a single no-op file path that OSS users can use here so they don't need to come up with one (e.g., `./no-op` or whatever passes validation).

Comment (ptgott) [inline docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx:212]:
```suggestion
(OSS users can provide any valid local file path here - it isn't used by the Auth Service in a Teleport OSS install.)
```

File: docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx [hunk 7]
```diff
@@ -0,0 +1,714 @@
...
+  which is useful for customers deploying Teleport in regulated environments. Configuration of object lock is out of
+  the scope of this guide.
+</Admonition>
+
+### Cluster domain
+
+The reference Terraform deployment sets the Teleport cluster up to be available on a domain defined in Route 53, referenced
+by the [`route53_domain`](#route53\_domain) variable. In our example this would be `teleport.example.com`.
+
+Teleport's web interface will be available on port 443 - [https://teleport.example.com](https://teleport.example.com).
```
Comment (ptgott) [inline docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx:401]: I wouldn't link to the example URL.

```suggestion
Teleport's web interface will be available on port 443 of your Teleport cluster host.
```

File: docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx [hunk 8]
```diff
@@ -0,0 +1,714 @@
...
+
+When using ACM, this is an A record aliased to the AWS Application Load Balancer.
+
+When using Let's Encrypt, this is an A record pointing to the instance's public IP.
+
+<Tabs>
```
Comment (ptgott) [inline docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx:407]: Same comments on this `Tabs` box as the similar `Tabs` box in the other guide

File: docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx [hunk 9]
```diff
@@ -0,0 +1,714 @@
...
+# Do you want to perform these actions?
+#  Terraform will perform the actions described above.
+#  Only 'yes' will be accepted to approve.
+
+#  Enter a value:
+```
+
+Entering `yes` here will start the Terraform deployment. It takes around 5 minutes to deploy in full.
+
+### Destroy/shutdown a Terraform deployment
```
Comment (ptgott) [inline docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx:503]:
```suggestion
### Destroy/shut down a Terraform deployment
```

File: docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx [hunk 10]
```diff
@@ -0,0 +1,714 @@
...
+You can see this after deploy by running `terraform output -raw cluster_web_address`:
+
+```code
+$ terraform output -raw cluster_web_address
+https://teleport.example.com
+```
+
+### Adding an admin user to the Teleport cluster
+
+To add users to the Teleport cluster, you will need to connect to a Teleport auth server via SSH and run the `tctl` command.
```
Comment (ptgott) [inline docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx:520]:
```suggestion
To add users to the Teleport cluster, you will need to connect to a Teleport Auth Service via SSH and run the `tctl` command.
```

File: docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx [hunk 11]
```diff
@@ -0,0 +1,714 @@
...
+
+1 - Use the key name and server IP to SSH into the auth server. Make sure that the AWS keypair that you specified in the
```
Comment (ptgott) [inline docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx:522]: We should use a standard Markdown ordered list here:

```
1. Item
1. Item
1. Item
```

Comment (ptgott) [inline docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx:522]:
```suggestion
1 - Use the key name and server IP to SSH into the Auth Service. Make sure that the AWS keypair that you specified in the
```

**Review by ptgott — APPROVED** [review]

File: docs/pages/installation.mdx [hunk 12]
```diff
@@ -364,31 +364,31 @@ from a trusted CA, e.g., Let's Encrypt.
...
-pre-installed. 
+pre-installed.
 
-These images are primarily intended for deploying a Teleport cluster using our 
+These images are primarily intended for deploying a Teleport cluster using our
 [reference Terraform code](https://github.com/gravitational/teleport/tree/branch/v(=teleport.major_version=)/examples/aws/terraform).
 
-See the [Starter Cluster Terraform](https://github.com/gravitational/teleport/tree/branch/v(=teleport.major_version=)/examples/aws/terraform/starter-cluster)
-and the [Running Teleport Enterprise in High Availability mode on AWS](deploy-a-cluster/deployments/aws-terraform.mdx)
+See the [Starter Cluster Terraform](deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx)
```
Comment (webvictim) [inline docs/pages/installation.mdx:372]:
```suggestion
See the [AWS Single-Instance Deployment](deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx)
```

File: docs/pages/deploy-a-cluster/deployments.mdx [hunk 13]
```diff
@@ -7,8 +7,10 @@ layout: tocless-doc
 These guides show you how to set up a full self-hosted Teleport deployment on
 the platform of your choice.
 
-- [AWS Terraform](./deployments/aws-terraform.mdx): Deploy HA Teleport with
-  Terraform Provider on AWS.
+- [AWS High Availability deployment with Terraform](./deployments/aws-ha-autoscale-cluster-terraform.mdx): Deploy HA Teleport with
```
Comment (webvictim) [inline docs/pages/deploy-a-cluster/deployments.mdx:10]:
```suggestion
- [AWS High Availability Deployment with Terraform](./deployments/aws-ha-autoscale-cluster-terraform.mdx): Deploy HA Teleport with
```

File: docs/pages/deploy-a-cluster/deployments/aws-ha-autoscale-cluster-terraform.mdx [hunk 14]
```diff
@@ -347,25 +406,48 @@ In our example this would be `s3://example-cluster/records`
...
 
-After deploying, you can get the hostname of the proxy/tunnel network load balancer if needed with this command:
+  - Teleport's proxy SSH interface will be available via a network load balancer with an AWS-controlled hostname on port 3023.
+  This is the default port used when connecting with the `tsh` client and will not require any additional configuration.
 
-```code
-$ aws elbv2 describe-load-balancers --names "${TF_VAR_cluster_name}-proxy" --query "LoadBalancers[*].DNSName" --output text
-# example-cluster-proxy-7c97b76593d6bf21.elb.us-east-1.amazonaws.com
-```
+  - Teleport's tunnel interface will be available via the same network load balancer with
```
[the remark this answers is not in the document]
Comment (webvictim) [reply docs/pages/deploy-a-cluster/deployments/aws-ha-autoscale-cluster-terraform.mdx:427]: https://goteleport.com/docs/reference/networking/#ports refers to reverse tunnels a number of times, so I've rephrased this to refer to the reverse tunnel listener of the Teleport Proxy Service.

File: docs/pages/deploy-a-cluster/deployments/aws-ha-autoscale-cluster-terraform.mdx [hunk 15]
```diff
@@ -347,25 +406,48 @@ In our example this would be `s3://example-cluster/records`
...
+  an AWS-controlled hostname on port 3024. This allows trusted clusters and nodes
+  connected via node tunneling to access the cluster.
+
+  - Teleport's Kubernetes interface will be available via the same network load balancer with
```
[the remark this answers is not in the document]
Comment (webvictim) [reply docs/pages/deploy-a-cluster/deployments/aws-ha-autoscale-cluster-terraform.mdx:431]: It's the Proxy Service's Kubernetes listener - clarified.

File: docs/pages/deploy-a-cluster/deployments/aws-ha-autoscale-cluster-terraform.mdx [hunk 16]
```diff
@@ -437,25 +519,31 @@ If you need to tear down a running deployment for any reason, you can run `terra
...
+$ terraform output -raw cluster_web_address
+https://teleport.example.com
+```
 
 ### Adding an admin user to the Teleport cluster
 
 To add users to the Teleport cluster, you will need to connect to a Teleport auth server via SSH and run the `tctl` command.
 
-1 - Use the AWS cli to get the IP of the bastion server:
+1 - Get the public IP of the bastion server using `terraform output -raw bastion_ip_public`:
```
[the remark this answers is not in the document]
Comment (webvictim) [reply docs/pages/deploy-a-cluster/deployments/aws-ha-autoscale-cluster-terraform.mdx:535]: I always find this to be horrifically inconsistent which is why I haven't done it previously. I'll endeavour to make it work here.

[the remark this answers is not in the document]
Comment (webvictim) [reply docs/pages/deploy-a-cluster/deployments/aws-ha-autoscale-cluster-terraform.mdx:535]: Same thing happens even when I indent each section by two spaces and leave the `1.` as the leftmost entry.

[the remark this answers is not in the document]
Comment (webvictim) [reply docs/pages/deploy-a-cluster/deployments/aws-ha-autoscale-cluster-terraform.mdx:535]: Think I've got it now. More spaces were required.

File: docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx (hunk 2 above)
Comment (webvictim) [reply docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx:36]: We already state this 20 lines below when we talk about `awscli` - was this intentional?

File: docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx (hunk 4 above)
Comment (webvictim) [reply docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx:209]: This was inappropriate here anyway as it's a single-instance deployment - fixed.

**Comment by webvictim** [comment]
Thanks for the huge review @ptgott!

File: examples/aws/terraform/ha-autoscale-cluster/connect.sh [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+    INSTANCE_ID=$2
+else
+    INSTANCE_ID="0"
+fi
+
+# shellcheck disable=SC2155
+export BASTION_IP=$(terraform output -raw bastion_ip_public)
+echo "Bastion IP: ${BASTION_IP?}"
+# shellcheck disable=SC2155
+export CLUSTER_NAME=$(terraform output -raw cluster_name)
```
Comment (mdwn) [inline examples/aws/terraform/ha-autoscale-cluster/connect.sh:24]:
```suggestion
CLUSTER_NAME="$(terraform output -raw cluster_name)"
```

File: examples/aws/terraform/ha-autoscale-cluster/connect.sh [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+echo "Cluster name: ${CLUSTER_NAME?}"
+
+if [[ "${INSTANCE_TYPE?}" == "auth" ]]; then
+    # shellcheck disable=SC2155
+    export SERVER_IP=$(aws ec2 describe-instances --filters "Name=tag:TeleportCluster,Values=${CLUSTER_NAME?}" "Name=tag:TeleportRole,Values=auth" --query "Reservations[${INSTANCE_ID?}].Instances[*].PrivateIpAddress" --output text)
```
Comment (mdwn) [inline examples/aws/terraform/ha-autoscale-cluster/connect.sh:29]:
```suggestion
    SERVER_IP="$(aws ec2 describe-instances --filters "Name=tag:TeleportCluster,Values=${CLUSTER_NAME?}" "Name=tag:TeleportRole,Values=auth" --query "Reservations[${INSTANCE_ID?}].Instances[*].PrivateIpAddress" --output text)"
```

File: examples/aws/terraform/ha-autoscale-cluster/connect.sh [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+    echo "Auth ${INSTANCE_ID?} IP: ${SERVER_IP?}"
+elif [[ "${INSTANCE_TYPE?}" == "proxy" ]]; then
+    # shellcheck disable=SC2155
+    export SERVER_IP=$(aws ec2 describe-instances --filters "Name=tag:TeleportCluster,Values=${CLUSTER_NAME?}" "Name=tag:TeleportRole,Values=proxy" --query "Reservations[${INSTANCE_ID?}].Instances[*].PrivateIpAddress" --output text)
```
Comment (mdwn) [inline examples/aws/terraform/ha-autoscale-cluster/connect.sh:33]:
```suggestion
    SERVER_IP="$(aws ec2 describe-instances --filters "Name=tag:TeleportCluster,Values=${CLUSTER_NAME?}" "Name=tag:TeleportRole,Values=proxy" --query "Reservations[${INSTANCE_ID?}].Instances[*].PrivateIpAddress" --output text)"
```

File: examples/aws/terraform/ha-autoscale-cluster/connect.sh [hunk 20]
```diff
@@ -0,0 +1,44 @@
...
+    echo "Proxy ${INSTANCE_ID?} IP: ${SERVER_IP?}"
+elif [[ "${INSTANCE_TYPE?}" == "node" ]]; then
+    # shellcheck disable=SC2155
+    export SERVER_IP=$(aws ec2 describe-instances --filters "Name=tag:TeleportCluster,Values=${CLUSTER_NAME?}" "Name=tag:TeleportRole,Values=node" --query "Reservations[*].Instances[*].PrivateIpAddress" --output text)
+    echo "Node IP: ${SERVER_IP?}"
+fi
+
+KEYPAIR_NAME=$(terraform output -raw key_name)
```
Comment (mdwn) [inline examples/aws/terraform/ha-autoscale-cluster/connect.sh:41]:
```suggestion
KEYPAIR_NAME="$(terraform output -raw key_name)"
```

File: examples/aws/terraform/ha-autoscale-cluster/connect.sh [hunk 21]
```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
```
Comment (mdwn) [inline examples/aws/terraform/ha-autoscale-cluster/connect.sh:1]:
```suggestion
#!/bin/bash

set -e
```

**Review by mdwn — APPROVED** [review]
A few nits wit the `connect.sh` script, but otherwise LGTM.

File: examples/aws/terraform/ha-autoscale-cluster/connect.sh [hunk 22]
```diff
@@ -0,0 +1,41 @@
...
+set -e
+# This script can be used to connect to instances behind the SSH bastion. It makes the assumption that either:
+# - The required keypair .pem file is in the same directory as this script
+# - The required keypair has been added to the running ssh-agent with `ssh-add /path/to/keypair.pem`
+# Example usage (server IDs are zero-indexed)
+# Connect to the first auth server in the ASG: ./connect.sh auth 0
+# Connect to the second proxy server in the ASG: ./connect.sh proxy 1
+# Connect to deployed example node: ./connect.sh node
+if [[ "$1" != "" ]]; then
+    INSTANCE_TYPE=$1
```
Comment (codingllama) [inline examples/aws/terraform/ha-autoscale-cluster/connect.sh:12]:
```suggestion
    INSTANCE_TYPE="$1"
```

**Review by codingllama — APPROVED** [review]
bot

**Comment by codingllama** [comment]
Would you mind addressing the comments before queueing?

**Pull request merged**